src/viam/trajex/totg/waypoint_accumulator.cpp

```diff
@@ -0,0 +1,79 @@
+#include <viam/trajex/totg/waypoint_accumulator.hpp>
+
+#include <stdexcept>
+
+namespace viam::trajex::totg {
+
+waypoint_accumulator::waypoint_accumulator(const xt::xarray<double>& waypoints) {
+    if (waypoints.dimension() != 2) {
+        throw std::invalid_argument{"Waypoints must be 2-dimensional"};
+    }
+    if (waypoints.shape()[0] == 0) {
+        throw std::invalid_argument{"Waypoints cannot be empty"};
+    }
```

Comment:
we can't start from an empty waypoints because we wouldn't be able to validate the dof?

mainly asking because i was wondering if we could start with an empty waypoints, and just use `add_waypoints` to populate the accumulator, but I got no idea if that would even be useful

Reply:
Roughly, yeah, re DoF. Also, if you have zero waypoints, there's sort of nothing to do anyway? I guess it wouldn't be that much more work to allow creating an empty waypoints accumulator. But I'm not sure it is very helpful to do so.